src/services/uniswap.js

```diff
@@ -0,0 +1,72 @@
+import Ethereum from '../services/eth';
+const uni = require('@uniswap/sdk')
+const ethers = require('ethers')
+const proxyArtifact = require('../static/uniswap_v2_router_abi.json')
+const debug = require('debug')('router')
+
+const GAS_LIMIT = 1200000  //120000000
+
+export default class Uniswap {
+  constructor (network = 'kovan') {
+    // network defaults to KOVAN
+    const providerUrl = process.env.ETHEREUM_RPC_URL
+    this.network = process.env.UNISWAP_NETWORK
+    this.chainID = uni.ChainId.KOVAN
+    this.provider = new ethers.providers.JsonRpcProvider(providerUrl)
+    this.exchangeProxy = process.env.UNISWAP_PROXY
+
+    if (network === 'kovan') {
+      this.chainID = uni.ChainId.KOVAN
+    } else if (network === 'mainnet') {
+      this.chainID = uni.ChainId.MAINNET
+    } else {
+      throw Error(`Invalid network ${network}`)
+    }
+  }
+
+  async fetch_route(tokenIn, tokenOut){
+      var route, pair, pairOne, pairTwo
+      var tIn = await uni.Fetcher.fetchTokenData(this.chainID, tokenIn)
+      var tOut = await uni.Fetcher.fetchTokenData(this.chainID, tokenOut)
+
+      try {
+      pair = await uni.Fetcher.fetchPairData(tIn, tOut)
+      route = new uni.Route([pair], tIn, tOut)
+      }
+      catch(err){
+      console.log('Trying alternative/indirect route.')
+      pairOne = await uni.Fetcher.fetchPairData(tIn, uni.WETH[this.chainID])
+      pairTwo = await uni.Fetcher.fetchPairData(tOut, uni.WETH[this.chainID])
+      route = new uni.Route([pairOne, pairTwo], tIn, tOut)
+      }
+      return route
+  }
+
+  async swapExactIn (wallet, route, tokenAddress, amountIn, gasPrice) {
```

Comment:
Similar to Balancer, I think we need a `swapExactOut` function to handle buy trades. For buy trades we need to calculate the input token amount given a certain output token amount. Per this [docs page](https://uniswap.org/docs/v2/SDK/trade/), we should use the `TradeOut` trade type to do this. 

FYI I recommend a separate function instead of handling both cases in a single `swap` function - I tried the latter initially with Balancer and it got really confusing.